### PR TITLE
Add chainer.as_variable

### PR DESCRIPTION
**Pull request by beam2d**
This PR adds a convenient function `chainer.to_variable`. It can be used to force a value having a type `Variable`. This function is designed to be used in differentiable functions that may skip computation; in such a case, these functions have to conditionally wrap an array by `Variable`. The `to_variable` can be used in such a case. Note that `requires_grad` is set to `False` because of assuming such usage.

### CODE REVIEW

File: chainer/functions/connection/linear.py [hunk 1]
```diff
@@ -100,3 +100,4 @@ def linear(x, W, b=None):
         return LinearFunction()(x, W)
     else:
         return LinearFunction()(x, W, b)
+0
```
Comment (okuta) [inline chainer/functions/connection/linear.py]: Please remove this line

**Comment by beam2d** [comment]
Oops, fixed that part. Also rebased to the master.

**Comment by niboshi** [comment]
How about `as_variable` instead, because it returns `Variable` as-is?

**Comment by beam2d** [comment]
That sounds better to me. One concern is an inconsistency with the existing names like `to_cpu` and `to_gpu`, but I think `as_variable` better explains what it does. I will fix it.

**Comment by beam2d** [comment]
Updated.

**Comment by okuta** [comment]
Please fix import order.

**Comment by beam2d** [comment]
Fixed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**